### PR TITLE
Simplify assertion same result but simplest.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/linear/BigRealTest.java [hunk 1]
```diff
@@ -167,7 +167,7 @@ public void testEqualsAndHashCode() {
         BigReal nullReal = null;
         Assert.assertEquals(zero, zero);
         Assert.assertNotEquals(zero, nullReal);
-        Assert.assertFalse(zero.equals(Double.valueOf(0)));
+        Assert.assertNotEquals(zero, Double.valueOf(0), 0.0);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/linear/BigRealTest.java:170]: This is testing the `equals()` method. Please revert.

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java [hunk 2]
```diff
@@ -259,7 +259,7 @@ public void testEqualsAndHashCode() {
         int emptyHash = u.hashCode();
         Assert.assertEquals(u, u);
         Assert.assertNotEquals(u, t);
-        Assert.assertFalse(u.equals(Double.valueOf(0)));
+        Assert.assertNotEquals(u, Double.valueOf(0), 0.0);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java:262]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java [hunk 3]
```diff
@@ -271,8 +271,8 @@ public void testEqualsAndHashCode() {
         u.addValue(new double[] { 3d, 1d });
         u.addValue(new double[] { 4d, 1d });
         u.addValue(new double[] { 5d, 1d });
-        Assert.assertFalse(t.equals(u));
-        Assert.assertFalse(u.equals(t));
+        Assert.assertNotEquals(t, u);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java:274]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java [hunk 4]
```diff
@@ -281,15 +281,15 @@ public void testEqualsAndHashCode() {
         t.addValue(new double[] { 3d, 1d });
         t.addValue(new double[] { 4d, 1d });
         t.addValue(new double[] { 5d, 1d });
-        Assert.assertTrue(t.equals(u));
-        Assert.assertTrue(u.equals(t));
+        Assert.assertEquals(t, u);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java:284]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java [hunk 5]
```diff
@@ -281,15 +281,15 @@ public void testEqualsAndHashCode() {
...
+        Assert.assertEquals(u, t);
         Assert.assertEquals(u.hashCode(), t.hashCode());
 
         // Clear and make sure summaries are indistinguishable from empty summary
         u.clear();
         t.clear();
-        Assert.assertTrue(t.equals(u));
-        Assert.assertTrue(u.equals(t));
+        Assert.assertEquals(t, u);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/MultivariateSummaryStatisticsTest.java:291]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/ResizableDoubleArrayTest.java [hunk 6]
```diff
@@ -452,11 +452,11 @@ public void testEqualsAndHashCode() throws Exception {
         // Wrong type
         ResizableDoubleArray first = new ResizableDoubleArray();
         Double other = Double.valueOf(2);
-        Assert.assertFalse(first.equals(other));
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/ResizableDoubleArrayTest.java:455]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/ResizableDoubleArrayTest.java [hunk 7]
```diff
@@ -452,11 +452,11 @@ public void testEqualsAndHashCode() throws Exception {
...
+        Assert.assertNotEquals(first, other, 0.0);
 
         // Null
         other = null;
-        Assert.assertFalse(first.equals(other));
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/ResizableDoubleArrayTest.java:459]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/rank/PSquarePercentileTest.java [hunk 8]
```diff
@@ -124,8 +124,8 @@ public void testCopyConsistencyWithInitialFirstFewElements() {
                 (int) (testArray.length - index));
         // Check same
         // Explicit test of the equals method
-        Assert.assertTrue(master.equals(master));
-        Assert.assertTrue(replica.equals(replica));
+        Assert.assertEquals(master, master);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/rank/PSquarePercentileTest.java:127]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/SummaryStatisticsTest.java [hunk 9]
```diff
@@ -161,7 +161,7 @@ public void testEqualsAndHashCode() {
         int emptyHash = u.hashCode();
         Assert.assertEquals("reflexive", u, u);
         Assert.assertNotEquals("non-null compared to null", u, t);
-        Assert.assertFalse("wrong type", u.equals(Double.valueOf(0)));
+        Assert.assertNotEquals("wrong type", u, Double.valueOf(0), 0.0);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/SummaryStatisticsTest.java:164]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/StorelessUnivariateStatisticAbstractTest.java [hunk 10]
```diff
@@ -80,7 +80,7 @@ public void testEqualsAndHashCode() {
             (StorelessUnivariateStatistic) getUnivariateStatistic();
         StorelessUnivariateStatistic statistic2 = null;
 
-        Assert.assertFalse("non-null, compared to null", statistic.equals(statistic2));
+        Assert.assertNotEquals("non-null, compared to null", statistic, statistic2);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/StorelessUnivariateStatisticAbstractTest.java:83]: This is testing `equals()`, please revert

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/StatisticalSummaryValuesTest.java [hunk 11]
```diff
@@ -35,7 +35,7 @@ public void testEqualsAndHashCode() {
         StatisticalSummaryValues t = null;
         Assert.assertEquals("reflexive", u, u);
         Assert.assertNotEquals("non-null compared to null", u, t);
-        Assert.assertFalse("wrong type", u.equals(Double.valueOf(0)));
+        Assert.assertNotEquals("wrong type", u, Double.valueOf(0), 0.0);
```
Comment (aherbert) [inline commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/StatisticalSummaryValuesTest.java:38]: This is testing `equals()`, please revert

File: commons-math-neuralnet/src/test/java/org/apache/commons/math4/neuralnet/NeuronTest.java [hunk 12]
```diff
@@ -92,7 +92,7 @@ public void testCopy() {
         final Neuron copy = n.copy();
 
         // Check that original and copy have the same value.
-        Assert.assertTrue(n.getFeatures()[0] == copy.getFeatures()[0]);
+        Assert.assertEquals(n.getFeatures()[0], copy.getFeatures()[0], 0.0);
```
Comment (aherbert) [inline commons-math-neuralnet/src/test/java/org/apache/commons/math4/neuralnet/NeuronTest.java:95]: No delta required when delta==0.0

**Review by aherbert — CHANGES_REQUESTED** [review]
A load of changes here in methods named testEqualsAndHashCode. These must test the equals method to get path coverage. So please revert them.

This also fails CI on Java 8 for a compile error. Please check that.

**Comment by arturobernalg** [comment]
@aherbert 
Done.

File: commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/StorelessUnivariateStatisticAbstractTest.java (hunk 10 above)
Comment (aherbert) [reply commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/stat/descriptive/StorelessUnivariateStatisticAbstractTest.java:83]: This still needs to be reverted. Thanks.

**Pull request closed without merging**